### PR TITLE
Move shortcut widget to feature module

### PR DESCRIPTION
**Pull request by wmontwe**
As part of #7933, this moves the `LauncherShortcut` to a the feature module. Translations have been moved to decouple it from the legacy code base and K-9 is replaced with `AppNameProvider`

The `LauncherShortcut` is technically not really a widget, but behaves as one and will be listed as app widget. Therefore I moved it under the widget namespace. This is important as it has shared functionality with the `MessageListWidget`.

Tasks once reviewed:

- [ ] Lock Weblate
- [ ] Pull latest translations from weblate, check if moved resources are affected and update this branch
- [ ] After Merge update weblate to pickup the new module
- [ ] Unlock Weblate

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -69,6 +69,7 @@ include(
 
 include(
     ":feature:widget:message-list",
+    ":feature:widget:shortcut",
```
Comment (cketti) [inline settings.gradle.kts:72]: I think we should avoid using the generic name "shortcut".

We only have one type of shortcut now. But could have multiple ones in the future, e.g. shortcuts to open a compose dialog with a particular sender identity and a given recipient address.

So we could make this `:feature:widget:account-shortcut` or `:feature:shortcut:account`.

File: feature/widget/shortcut/src/main/kotlin/app/k9mail/feature/widget/shortcut/LauncherShortcutActivity.kt [hunk 2]
```diff
@@ -9,15 +9,20 @@ import androidx.annotation.AttrRes
...
 import app.k9mail.core.ui.legacy.theme2.common.R as CommonR
 
-class LauncherShortcuts : AccountList() {
+class LauncherShortcutActivity : AccountList() {
+    private val appNameProvider: AppNameProvider by inject()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setTitle(R.string.shortcuts_title)
+        setTitle(resources.getString(R.string.shortcut_widget_title, appNameProvider.appName))
```
Comment (cketti) [inline feature/widget/shortcut/src/main/kotlin/app/k9mail/feature/widget/shortcut/LauncherShortcutActivity.kt:25]: Having the app name here feels unnecessary. I think it's fine to just use "Accounts" for the activity title.

**Review by cketti — APPROVED** [review]

File: feature/widget/shortcut/src/main/kotlin/app/k9mail/feature/widget/shortcut/LauncherShortcutActivity.kt (hunk 2 above)
Comment (wmontwe) [reply feature/widget/shortcut/src/main/kotlin/app/k9mail/feature/widget/shortcut/LauncherShortcutActivity.kt:25]: At the moment all widgets use the app name for their configuration interface. We could of course entirely remove the app name.

File: settings.gradle.kts (hunk 1 above)
Comment (wmontwe) [reply settings.gradle.kts:72]: While there might be different shortcut types eventually. They all need to share the same maximum limit of shortcuts allowed by the system. I don't see a reason to further split this up into individual shortcut modules, until we have a concrete use case.

File: feature/widget/shortcut/src/main/kotlin/app/k9mail/feature/widget/shortcut/LauncherShortcutActivity.kt (hunk 2 above)
Comment (wmontwe) [reply feature/widget/shortcut/src/main/kotlin/app/k9mail/feature/widget/shortcut/LauncherShortcutActivity.kt:25]: As the translations are using a form of `K-9 Accounts` I need to drop the translations then, as they are not correct anymore.

Comment (cketti) [reply feature/widget/shortcut/src/main/kotlin/app/k9mail/feature/widget/shortcut/LauncherShortcutActivity.kt:25]: :+1:

**Pull request merged**